### PR TITLE
nixos/klipper: Tweaks

### PR DESCRIPTION
**Pull request by zhaofengli**
###### Motivation for this change

Building a new printer, and this time it's powered by NixOS and not Yocto :stuck_out_tongue:

Anyways, this PR extends `services.klipper` to facilitate the use of alternative integrations (e.g., Moonraker). New options are added to allow the configuration of `--input-tty`, `--api-server`, as well as the user/group Klipper will run as.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130616` at f8658c81 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>klipper</li>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md">name-and-version</a>

Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/servers/klipper/default.nix:8:3:
```
  |
8 |   name = "klipper";
  |   ^
```

Near pkgs/servers/klipper/default.nix:9:3:
```
  |
9 |   version = "unstable-2021-07-15";
  |   ^
```
</li>
  </ul>
</details>

File: nixos/modules/services/misc/klipper.nix [hunk 1]
```diff
@@ -17,6 +17,26 @@ in
...
+      group = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Group account under which Klipper runs.
+
+          If null is specified (default), a temporary user will be created by systemd.
+        '';
+      };
+
```
Comment (lovesegfault) [inline nixos/modules/services/misc/klipper.nix:57]: I think it's best to just default to a `klipper` DynamicUser.

**Review by lovesegfault — APPROVED** [review]
LGTM overall, left a small nit.

It'd be nice if modules were added for moonraker and the like.

Thanks for the changes!

**Comment by lovesegfault** [comment]
Ah, I now see you made PR's for that too.

Can you add me as a reviewer in this series of changes? Definitely interested in improving the Klipper experience in NixOS :)

**Comment by zhaofengli** [comment]
> Ah, I now see you made PR's for that too.
> 
> Can you add me as a reviewer in this series of changes? Definitely interested in improving the Klipper experience in NixOS :)

Sure thing! I wanted to add you to both PRs but I forgot.

File: nixos/modules/services/misc/klipper.nix (hunk 1 above)
Comment (zhaofengli) [reply nixos/modules/services/misc/klipper.nix:57]: If it's `null` then it will use `DynamicUser` same as the current behavior.

**Review by lovesegfault — APPROVED** [review]

**Pull request merged**